### PR TITLE
test: adding ica test for multiple controllers, single host

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Additional test for #791 #803 

Requires #785 so opening as draft

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/ibc_module_test.go [hunk 1]
```diff
@@ -632,3 +636,69 @@ func (suite *InterchainAccountsTestSuite) TestOnTimeoutPacket() {
...
+
+			accAddressChainA, found := suite.chainB.GetSimApp().ICAHostKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), pathAToB.EndpointB.ConnectionID, pathAToB.EndpointA.ChannelConfig.PortID)
+			suite.Require().True(found)
+
+			accAddressChainC, found := suite.chainB.GetSimApp().ICAHostKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), pathCToB.EndpointB.ConnectionID, pathCToB.EndpointA.ChannelConfig.PortID)
+			suite.Require().True(found)
+
+			suite.Require().NotEqual(accAddressChainA, accAddressChainC)
+
+			// TODO: Update icatypes.PortID to pathAToB.EndpointA.ChannelConfig.PortID
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/ibc_module_test.go:693]: These TODOs are pending #785

**Review by seantking — APPROVED** [review]
:handshake:

File: modules/apps/27-interchain-accounts/controller/ibc_module_test.go [hunk 2]
```diff
@@ -632,3 +636,67 @@ func (suite *InterchainAccountsTestSuite) TestOnTimeoutPacket() {
...
+			// Setup a new path from C(controller) -> B(host)
+			// NOTE: Here the version metadata is overridden to include to the next host connection sequence (i.e. chainB's connection to chainC)
+			nextConnectionSeq := suite.chainB.App.GetIBCKeeper().ConnectionKeeper.GetNextConnectionSequence(suite.chainB.GetContext())
+			TestVersion = string(icatypes.ModuleCdc.MustMarshalJSON(&icatypes.Metadata{
+				Version:                icatypes.Version,
+				ControllerConnectionId: ibctesting.FirstConnectionID,
+				HostConnectionId:       connectiontypes.FormatConnectionIdentifier(nextConnectionSeq),
+			}))
+			pathCToB = NewICAPath(suite.chainC, suite.chainB)
+			suite.coordinator.SetupConnections(pathCToB)
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_module_test.go:678]: probably could have reordered this so you could use pathCToB.Endpoint for the connection sequence

**Review by colin-axner — APPROVED** [review]
Thanks!

File: modules/apps/27-interchain-accounts/controller/ibc_module_test.go (hunk 2 above)
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/ibc_module_test.go]: Yea nice catch, updated with your suggestion, thanks!

**Pull request merged**